src/main/java/net/dries007/tfc/objects/te/TEAnvilTFC.java

```diff
@@ -240,7 +241,9 @@ public void addStep(@Nonnull EntityPlayer player, @Nullable ForgeStep step)
             if (recipe != null)
             {
                 AnvilRecipe completedRecipe = recipe; // Hold the current recipe, as setting the input slot to empty will clear it
-                if (workingProgress == workingTarget && completedRecipe.matches(steps))
+                int workMin = workingTarget - ConfigTFC.General.DIFFICULTY.acceptableAnvilRange;
+                int workMax = workingTarget + ConfigTFC.General.DIFFICULTY.acceptableAnvilRange;
+                if ((workingProgress < workMax && workingProgress > workMin) && completedRecipe.matches(steps))
```

Comment:
```suggestion
                if ((workingProgress <= workMax && workingProgress >= workMin) && completedRecipe.matches(steps))
```
otherwise, this will make it impossible to finish if both `workMin` and `workMax` are the same (default config = 0)